### DISCUSSION TITLE
Script Executor example: Distributable Version Approach?

### DISCUSSION THREAD

**Discussion by LeoM1970** [zauberzeug/nicegui#3871] [Q&A]
I made a variation of your 'Script Executor' example: https://github.com/zauberzeug/nicegui/blob/main/examples/script_executor/main.py

This allows me to launch a few of my scripts within the same NiceGUi environment, which is great.

Now I would like to share a distributable version of it... But is this a terrible idea?  I am struggling to think of an approach that will work.

When packaging the script "as is" into a one-file executable, the behaviour is as follows:

Action: User clicks on a button to launch a script (i.e. launching: run_command('python3 hello.py')
Result: The contextual window displays NiceGUI (apparently) re-launching itself? - e.g.:  "NiceGUI ready to go on http://local host:8002"

Is it feasible to modify the script executor in a manner that it will work within a distributable version?

If not, could you suggest alternative ideas? I suppose I could write a GI that launches other .exe versions of scripts? This, however, will miss the all-in-one contextual window of the 'Script Executor'.

Many thanks in advance.

**Comment by falkoschindler** [comment]
@LeoM1970 One thing that pops into my mind is to call `ui.run` with `reload=False`. This might prevent the app from reloading.
But there might be other caveats when packaging an app with additional scripts.
